### PR TITLE
add tests convert functions

### PR DESCRIPTION
**Pull request by ortex**
#45

### CODE REVIEW

File: roc/convert_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		{false, 0},
+	}
+	for _, tt := range tests {
+		t.Run(strconv.FormatBool(tt.arg), func(t *testing.T) {
+			assert.Equal(t, tt.result, uint(go2cBool(tt.arg)))
+		})
+	}
+}
+
+func Test_go2cStr_c2goStr(t *testing.T) {
```
Comment (gavv) [inline roc/convert_test.go:25]: Could we also add tests for just c2goStr and just go2cStr?

The current tests don't actually check zero-byte handling, it would be nice to cover that part more precisely.

**Review by gavv — COMMENTED** [review]
Thanks for PR!

File: roc/convert_test.go (hunk 1 above)
Comment (ortex) [reply roc/convert_test.go]: added

File: roc/convert.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ import (
 	"C"
 )
 
+type char = C.char
```
Comment (gavv) [inline roc/convert.go:7]: nit: let's use `char` instead of `C.char` below

File: roc/convert_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func Test_c2goStr(t *testing.T) {
+	tests := []struct {
+		name   string
+		arg    []char
+		result string
+	}{
+		{name: "str", arg: []char{'s', 't', 'r', '\x00'}, result: "str"},
+		{name: "empty", arg: []char{'\x00'}, result: ""},
```
Comment (gavv) [inline roc/convert_test.go:49]: Let's add two more tests:

* two zero bytes at the end instead of one
* unexpected zero byte in the middle of the string

File: roc/convert_test.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+func Test_go2cStr(t *testing.T) {
+	tests := []struct {
+		name   string
+		arg    string
+		result []char
+	}{
+		{name: "str", arg: "str", result: []char{'s', 't', 'r', '\x00'}},
+		{name: "str00str", arg: "str\x00", result: []char{'s', 't', 'r', '\x00', '\x00'}},
+		{name: "empty", arg: "", result: []char{'\x00'}},
```
Comment (gavv) [inline roc/convert_test.go:33]: * let's rename `name: "str00str"` to `name: "str0str"` because there is only one zero byte
* let's add a test when zero byte is in the end

**Comment by gavv** [comment]
https://github.com/roc-streaming/roc-go/issues/50

File: roc/convert.go (hunk 2 above)
Comment (ortex) [reply roc/convert.go:7]: done

File: roc/convert_test.go (hunk 3 above)
Comment (ortex) [reply roc/convert_test.go]: done

File: roc/convert_test.go (hunk 4 above)
Comment (ortex) [reply roc/convert_test.go]: done

**Comment by gavv** [comment]
Thanks!
